src/choker.cpp

```diff
@@ -168,13 +158,24 @@ namespace {
 		//   |             V             |
 		//   +---------------------------+
 		//   0%    num have pieces     100%
-		int const t1_total = t1->torrent_file().num_pieces();
-		int const t2_total = t2->torrent_file().num_pieces();
-		int const score1 = (lhs->num_have_pieces() < t1_total / 2
-			? t1_total - lhs->num_have_pieces() : lhs->num_have_pieces()) * 1000 / t1_total;
-		int const score2 = (rhs->num_have_pieces() < t2_total / 2
-			? t2_total - rhs->num_have_pieces() : rhs->num_have_pieces()) * 1000 / t2_total;
+		std::shared_ptr<torrent> const t = peer->associated_torrent().lock();
+		TORRENT_ASSERT(t);
+
+		std::int64_t const total_size = t->torrent_file().total_size();
+		std::int64_t have_size = peer->num_have_pieces() * t->torrent_file().piece_length();
+		have_size = std::max(have_size, peer->uploaded_at_last_round());
+		return int(std::abs((have_size - total_size / 2) * 2000 / total_size));
```

Comment:
you changed the scale from `1000` to `2000`, was that deliberate? if so, what's the reason?

Reply:
To make the score falls into [0, 1000], neater then the origin [0, 500].